partiql-eval/src/eval.rs

```diff
@@ -46,56 +44,172 @@ pub enum EvaluationError {
     InvalidEvaluationPlan(String),
 }
 
+pub trait Evaluable: Debug {
+    fn evaluate(&mut self, ctx: &dyn EvalContext) -> Option<Value>;
+    fn update_input(&mut self, input: &Value);
+}
+
 #[derive(Debug)]
-pub struct Scan {
+pub struct EvalScan {
     pub expr: Box<dyn EvalExpr>,
     pub as_key: String,
+    pub at_key: String,
```

Comment:
Any reason this is modeled as a `String` rather than an `Option<String>`? I think modeling it as an optional type is less confusing to represent whether an `at_key` is provided or not (`None` rather than empty string for indicating empty string is used.

Reply:
I've actually tried it with `Option` as well but then found the API as being harder to work with; this is based on the assumption that in evaluation, we're in lower levels and initialization is being done via some other procedure, therefore IMO it's more ergonomic to use plain `String` for the code that wants to generate the plan. On the other hand, I think it makes sense to have it as optional in `LogicalPlan` the same as `Unpivot`. I don't have a strong opinion about this at this stage btw, let me know if you want to discuss about this further.

Reply:
Ah, got it. I guess to me, `Option` is "lower level" or less ambiguous than using `String`. I don't have a strong opinion on it either. We can always iterate as we learn more and adjust as needed.

Reply:
I would prefer `Option<String>` for `at_key`, since it is optional.

It should be easy enough to handle the optional:

In `EvalScan`, replace

```rust

if !at_key.is_empty() {
...
}

```

with

```rust

if let Some(at_key) = &self.at_key {
...
}

```


And for creation all the `"".to_string()`s become `None` and any `"at_key_name".to_string()` is `Some("at_key_name".into())`

Reply:
Fair, I hear you both, will change in the upcoming commit.